### PR TITLE
Added new error code for unhandled http responses

### PR DESCRIPTION
**Pull request by rohitnarula7176**
*Added new error code to deal with unhandled http responses (i.s 404's)

### CODE REVIEW

File: MSAL/src/public/MSALError.h [hunk 1]
```diff
@@ -94,8 +94,9 @@ typedef NS_ENUM(NSInteger, MSALErrorCode)
     
     MSALErrorInvalidRequest              = -42002,
     MSALErrorInvalidClient               = -42003,
-    MSALErrorInvalidGrant               = -42004,
-    MSALErrorInvalidScope               = -42005,
+    MSALErrorInvalidGrant                = -42004,
+    MSALErrorInvalidScope                = -42005,
+    MSALErrorUnhandledResponse           = -42006,
```
Comment (oldalton) [inline MSAL/src/public/MSALError.h]: would be nice to add a comment for what this error means to our developers

File: MSAL/src/requests/MSALSilentRequest.m [hunk 2]
```diff
@@ -187,7 +187,8 @@ - (BOOL)isServerUnavailable:(NSError *)error
         return NO;
     }
     
-    return ([error code] >= 500 && [error code] <= 599);
+    NSInteger errorCode = [[error.userInfo objectForKey:MSIDHTTPResponseCodeKey] intValue];
+    return errorCode >=500 && errorCode <=599;
```
Comment (oldalton) [inline MSAL/src/requests/MSALSilentRequest.m]: nit: spacing

File: MSAL/src/requests/MSALSilentRequest.m [hunk 3]
```diff
@@ -187,7 +187,8 @@ - (BOOL)isServerUnavailable:(NSError *)error
         return NO;
     }
     
-    return ([error code] >= 500 && [error code] <= 599);
+    NSInteger errorCode = [[error.userInfo objectForKey:MSIDHTTPResponseCodeKey] intValue];
```
Comment (oldalton) [inline MSAL/src/requests/MSALSilentRequest.m]: please also check for actual error code to be unhandled response for completeness

File: MSAL/src/requests/MSALSilentRequest.m (hunk 2 above)
Comment (rohitnarula7176) [reply MSAL/src/requests/MSALSilentRequest.m]: done

File: MSAL/src/public/MSALError.h (hunk 1 above)
Comment (rohitnarula7176) [reply MSAL/src/public/MSALError.h]: done

File: MSAL/src/requests/MSALSilentRequest.m (hunk 3 above)
Comment (rohitnarula7176) [reply MSAL/src/requests/MSALSilentRequest.m]: done

**Pull request merged**